src/org/sosy_lab/common/configuration/converters/FileTypeConverter.java

```diff
@@ -69,7 +69,7 @@
           File.class, Path.class, PathTemplate.class, PathCounterTemplate.class);
 
   private static final String TEMP_DIR =
-      StandardSystemProperty.JAVA_IO_TMPDIR.value() + File.separator;
+      Paths.get(StandardSystemProperty.JAVA_IO_TMPDIR.value() + File.separator).toString();
```

Comment:
This constant absolutely needs to have `File.separator` at its end, because we check for `path.startsWith(TEMP_DIR)`, which would wrongly accept paths outside the temp dir otherwise.

If a double slash is a problem here, it needs to be fixed somwhere else. But isn't this part of the change unnecessary anyway?

Reply:
In OpenBSD `StandardSystemProperty.JAVA_IO_TMPDIR.value()` evaluates to `"/tmp/"`. Then adding another `"/"` to it will result in a String `"/tmp//"`. Making it a Path object (and then using `toString()`) will remove duplicate `"/"` characters.
But I am not certain if on other systems the value from `Path.toString()` always ends with the separator.